119 add to strategy page list of tokens the strategy owns

sanvaldi opened the pull request:
- Adds new Tokens Tab in SingleStrategy page
- Renders all strategy tokens data (name, symbol, etherscan link, balance)
- Triggering fetch request in newly added StrategyTokens component

<img width="1727" alt="Screen Shot 2022-02-13 at 4 17 42 PM" src="https://user-images.githubusercontent.com/43012589/153770998-6815ac27-0a99-42df-9578-65e039c34bbb.png">


We're using https://api.ethplorer.io/ to get all strategy tokens since it seems it cannot be done using etherscan api as stated here: https://ethereum.stackexchange.com/questions/34335/etherscan-api-get-all-token-balances

A free api key is set in /utils/env.ts but we need to replace it for a production key  before merging.

The free key has usage limits:

Requests are limited to 2 per second, 30/min, 200/hour, 1000/24hours, 3000/week.
Max number of transactions/operations in response: 100
Max allowed timestamp age: 30 days

more info here: https://github.com/EverexIO/Ethplorer/wiki/Ethplorer-API

sanvaldi committed:
Added Tokens Tab in SingleStrategy

sanvaldi committed:
Added StrategyTokens component

Fetching strategy tokens from https://api.ethplorer.io/getAddressInfo/

sanvaldi committed:
Added TokenCard component

Using this component to render strategy tokens data

sanvaldi committed:
Added ethplorerKey to env.ts

TODO: replace freekey for production key

sanvaldi committed:
fixed typescript build errors

declared Token type and setting <Array<Token>> with useState

sanvaldi committed:
Merge remote-tracking branch 'original/master' into 119-Add-to-Strategy-Page-list-of-tokens-the-strategy-owns

sanvaldi committed:
Updated components using MUI v5

updated mui imports

sanvaldi committed:
Added loadingSpinner while fetching tokensData

edwinzhng commented on src/components/app/SingleStrategy/StrategyTokens.tsx:
```diff
@@ -0,0 +1,70 @@
+    // fetching strategy tokens
+    useEffect(() => {
+        fetch(
+            `https://api.ethplorer.io/getAddressInfo/${strategy.token.address}?apiKey=${ethplorerKey}`
```
I think you mean `strategy.address` here, otherwise you're getting the assets for the token contract, not the strategy contract

edwinzhng commented on src/components/app/SingleStrategy/StrategyTokens.tsx:
```diff
@@ -0,0 +1,70 @@
+        data.push({
+            name: tokensData[token].tokenInfo.name,
+            symbol: tokensData[token].tokenInfo.symbol,
+            balance: tokensData[token].balance,
```
This balance also likely needs to be formatted since the values being displayed are unreasonably big. eg) right now if you look at the strategy [0x79Be3edA1d7FE92c18B9539bce2611bf8dc9BFc5](https://yearn-watch-fibnhz5we-y-finance.vercel.app/network/ethereum/vault/0x6d765CbE5bC922694afE112C140b8878b9FB0390/strategy/0x79Be3edA1d7FE92c18B9539bce2611bf8dc9BFc5) which is fetching the tokens in the SUSHI contract by mistake, there should only be 3517 USD according to Etherscan (https://etherscan.io/address/0x6B3595068778DD592e39A122f4f5a5cF09C90fE2), not 3517736758

edwinzhng replied:
But if you look at a different token like FOX, there should only be 200, not 200000000000000000000. It is weird that the formatting is different for those tokens though, maybe the API returns some data on how to normalize it properly

Commit:
Fixed api request & handling no tokens found case

Replaced {strategy.token.address} with {strategy.address}.
Handling no tokens found case after timeout is complete and tokensData.length === 0

Commit:
Only rendering tokens with name

Some tokens with balance were retrieved from the request but didn't have name or symbol.
Checked etherscan and those are not rendered

Commit:
Fixed token balance

The API returns values in WEI for ether and also in the 18 decimal smallest unit for other tokens.
So balance / 10**18 gives the correct number

edwinzhng reviewed (approved):
data looks good to me now, I checked a few strategies manually. I will let @magic990619 take another look though. We should also figure out what to do for the API key, but we could create a separate issue to track this and address it later
